### DISCUSSION TITLE
How to get session id from getSession()?

### DISCUSSION THREAD

**Discussion by ong4mes** [nextauthjs/next-auth#3614] [Help]
I want to implement session management, using next-auth with the prisma plugin.
How can I get id of the current sessiion, so I know which is the current session? getSession() is only returning access token and user information.

I want to display all the users sessions and add a button to each to remove the session except the current one.

Thanks for the help

**Comment by soullessxo** [comment]
Session token stored as http-only cookie,  you can get it only server-side from request object `req.cookies['next-auth.session-token']`, for more convenient way dig deeper.

**Comment by rexfordessilfie** [comment]
Hi @ong4mes! I haven't fully attempted this before, but you can next-auth with the Prisma Database adapter, which will give you access to a `Session` model in particular which you can query at your disposal to retrieve (unexpired) sessions that are not expired on the server side through your own API routes, and expire them if you wish by updating the `expires` column to the current date to expire them.

Check out this [link](https://authjs.dev/reference/adapters#models) which discusses the models, and then how to integrate the [Prisma adapter](https://authjs.dev/reference/adapter/prisma)
